compiler/src/test/kotlin/io/spine/protodata/InsertionPointTest.kt

```diff
@@ -0,0 +1,117 @@
+ */
+
+package io.spine.protodata
+
+import com.google.common.truth.Truth.assertThat
+import com.google.protobuf.compiler.PluginProtos
+import io.spine.protodata.renderer.SourceFileSet
+import io.spine.protodata.renderer.codeLine
+import io.spine.protodata.test.CatOutOfTheBoxEmancipator
+import io.spine.protodata.test.KotlinInsertionPoint.FILE_END
+import io.spine.protodata.test.KotlinInsertionPoint.FILE_START
+import io.spine.protodata.test.KotlinInsertionPoint.LINE_FOUR_COL_THIRTY_THREE
+import io.spine.protodata.test.VariousKtInsertionPointsPrinter
+import java.nio.file.Path
+import kotlin.io.path.createFile
+import kotlin.io.path.div
+import kotlin.io.path.readLines
+import kotlin.io.path.writeText
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.io.TempDir
+
+class `Insertion points should` {
+
+    companion object {
+        private const val FILE_NAME = "sources.kt"
+    }
+
+    @Nested
+    inner class `appear at` {
+
+        private lateinit var pipeline: Pipeline
+        private lateinit var path: Path
+
+        @BeforeEach
+        fun preparePipeline(@TempDir path: Path) {
+            this.path = path
+            val file = path / FILE_NAME
+            file.createFile()
+            file.writeText("""
+             class LabMouse {
+                 companion object {
+                     const val I_AM_CONSTANT: String = "!!"
+                 }
+                
+                 fun letsHaveFun(): String {
+                     return "trololo"
+                 }
+            }
+            """.trimIndent())
+            pipeline = Pipeline(
+                listOf(),
+                listOf(VariousKtInsertionPointsPrinter(), CatOutOfTheBoxEmancipator()),
+                listOf(SourceFileSet.from(path)),
+                PluginProtos.CodeGeneratorRequest.getDefaultInstance(),
+            )
+        }
+
+        @Test
+        fun `the start of a file`() {
+            pipeline()
+            val file = path / FILE_NAME
+            val contents = file.readLines()
+            assertThat(contents)
+                .isNotEmpty()
+            assertThat(contents[0])
+                .contains(FILE_START.label)
+        }
+
+        @Test
+        fun `the end of a file`() {
+            pipeline()
+            val file = path / FILE_NAME
+            val contents = file.readLines()
+            assertThat(contents)
+                .isNotEmpty()
+            assertThat(contents.last())
+                .contains(FILE_END.label)
+        }
+
+        @Test
+        fun `a specific line and column`() {
+            pipeline()
+            val file = path / FILE_NAME
+            val contents = file.readLines()
+            assertThat(contents)
+                .isNotEmpty()
+            assertThat(contents[3])
+                .contains("I_AM_CONSTANT:/* ${LINE_FOUR_COL_THIRTY_THREE.codeLine} */      String")
```

Comment:
The insertion point name is a bit puzzling. What does it mean?

Reply:
Why do we have extra space before `String`?

Reply:
The extra space is there to account for all possible comment-closing syntax. Since when we insert code at the insertion points we don't know how the comment is formatted, we make sure to leave enough chars to wrap the insertion point label into a comment. The number of chars we pad is hard-coded to be 8.

This is the trick I was talking about in our call some time ago. An alternative would be to incorporate the column number into the insertion point text somehow, but that is also not very easy, since, as you pointed out, we would have to shift any other column numbers when inserting new code into a line.

The name of this test-only insertion point does not have a hidden meaning and reflects the place in the text where it's located: line 4, column 33.